# diff --git a/drivers/mtd/onenand/onenand_base.c b/drivers/mtd/onenand/onenand_base.c
# index 3b828fb..174384e 100644
# --- a/drivers/mtd/onenand/onenand_base.c
# +++ b/drivers/mtd/onenand/onenand_base.c
# #include <asm/errno.h> // extra line after asm/io.h
diff --git a/drivers/onenand/onenand_base.c b/drivers/onenand/onenand_base.c
index 3b828fb..174384e 100644
--- a/drivers/onenand/onenand_base.c
+++ b/drivers/onenand/onenand_base.c
@@ -20,6 +20,19 @@
 #include <asm/io.h>

 
+/* It should access 16-bit instead of 8-bit */
+static inline void *memcpy(void *dst, const void *src, unsigned int len)
+{
+ void *ret = dst;
+ short *d = dst;
+ const short *s = src;
+
+ len >>= 1;
+ while (len-- > 0)
+ *d++ = *s++;
+ return ret;
+}
+
 static const unsigned char ffchars[] = {
  0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
  0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, /* 16 */
